authority/provisioner/sshpop_test.go

```diff
@@ -0,0 +1,831 @@
+package provisioner
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/base64"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/assert"
+	"github.com/smallstep/certificates/db"
+	"github.com/smallstep/certificates/errs"
+	"github.com/smallstep/cli/crypto/pemutil"
+	"github.com/smallstep/cli/jose"
+	"golang.org/x/crypto/ssh"
+)
+
+func TestSSHPOP_Getters(t *testing.T) {
+	p, err := generateSSHPOP()
+	assert.FatalError(t, err)
+	id := "sshpop/" + p.Name
+	if got := p.GetID(); got != id {
+		t.Errorf("SSHPOP.GetID() = %v, want %v", got, id)
+	}
+	if got := p.GetName(); got != p.Name {
+		t.Errorf("SSHPOP.GetName() = %v, want %v", got, p.Name)
+	}
+	if got := p.GetType(); got != TypeSSHPOP {
+		t.Errorf("SSHPOP.GetType() = %v, want %v", got, TypeSSHPOP)
+	}
+	kid, key, ok := p.GetEncryptedKey()
+	if kid != "" || key != "" || ok == true {
+		t.Errorf("SSHPOP.GetEncryptedKey() = (%v, %v, %v), want (%v, %v, %v)",
+			kid, key, ok, "", "", false)
+	}
+}
+
+func createSSHCert(cert *ssh.Certificate, signer ssh.Signer) (*ssh.Certificate, *jose.JSONWebKey, error) {
+	jwk, err := jose.GenerateJWK("EC", "P-256", "ES256", "sig", "foo", 0)
+	if err != nil {
+		return nil, nil, err
+	}
+	cert.Key, err = ssh.NewPublicKey(jwk.Public().Key)
+	if err != nil {
+		return nil, nil, err
+	}
+	if err = cert.SignCert(rand.Reader, signer); err != nil {
+		return nil, nil, err
+	}
+	return cert, jwk, nil
+}
+
+func generateSSHPOPToken(p Interface, cert *ssh.Certificate, jwk *jose.JSONWebKey) (string, error) {
+	return generateToken("foo", p.GetName(), testAudiences.Sign[0], "",
+		[]string{"test.smallstep.com"}, time.Now(), jwk, withSSHPOPFile(cert))
+}
+
+func TestSSHPOP_authorizeToken(t *testing.T) {
+	key, err := pemutil.Read("./testdata/secrets/ssh_user_ca_key")
+	assert.FatalError(t, err)
+	signer, ok := key.(crypto.Signer)
+	assert.Fatal(t, ok, "could not cast ssh signing key to crypto signer")
+	sshSigner, err := ssh.NewSignerFromSigner(signer)
+	assert.FatalError(t, err)
+
+	type test struct {
+		p     *SSHPOP
+		token string
+		err   error
+		code  int
+	}
+	tests := map[string]func(*testing.T) test{
+		"fail/bad-token": func(t *testing.T) test {
+			p, err := generateSSHPOP()
+			assert.FatalError(t, err)
+			return test{
+				p:     p,
+				token: "foo",
+				code:  http.StatusUnauthorized,
+				err:   errors.New("sshpop.authorizeToken; error extracting sshpop header from token: extractSSHPOPCert; error parsing token: "),
+			}
+		},
+		"fail/error-revoked-db-check": func(t *testing.T) test {
+			p, err := generateSSHPOP()
+			assert.FatalError(t, err)
+			p.db = &db.MockAuthDB{
+				MIsSSHRevoked: func(sn string) (bool, error) {
+					return false, errors.New("fatal")
```

Comment:
Why new("fatal")?

Reply:
I'm guessing this is just a throwaway error?

Reply:
correct, throwaway error. I think I meant to use the word "force" here and got confused. I'll change it.